lib/clients/ZookeeperManager.js

```diff
@@ -0,0 +1,313 @@
+const zookeeper = require('node-zookeeper-client');
+const EventEmitter = require('events');
+const async = require('async');
+
+class ZookeeperManager extends EventEmitter {
+    /**
+     * Constructs an instance of the ZookeeperManager.
+     *
+     * @param {string} connectionString - The connection string for the ZooKeeper server.
+     * @param {Object} options - Configuration options for the ZooKeeper client.
+     * @param {Logger} log - An instance of a logging utility.
+     */
+    constructor(connectionString, options, log) {
+        super();
+
+        this.connectionString = connectionString;
+        this.log = log;
+        this.options = options;
+        this.client = null;
+
+        this._connect();
+    }
+
+    /**
+     * Establishes a connection to the ZooKeeper server.
+     *
+     * This method initializes a new ZooKeeper client using the provided connection string
+     * and options. It sets up event listeners for various client states such as 'connected',
+     * 'disconnected', and 'expired'. The method handles reconnection logic in case of
+     * client expiration and forwards ZooKeeper events through this class's EventEmitter interface.
+     *
+     * If 'autoCreateNamespace' is set in the options and the base path does not exist in ZooKeeper,
+     * this method tries to create the necessary path (namespace) in ZooKeeper.
+     *
+     * @method _connect
+     * @return {undefined}
+     */
+    _connect() {
+        // clean up exists client before reconnect
+        if (this.client) {
+            this.client.removeAllListeners();
+        }
+
+        this.client = zookeeper.createClient(this.connectionString, this.options);
+
+        this.client.once('connected', () => {
+            // TODO: ARTESCA-10337 The 'autoCreateNamespace' functionality is currently specific to
+            // Artesca and may be removed in future versions once the Zenko Operator can handle base path creation.
+            // Once removed, we can simply rely on the 'connected' state instead of the 'ready' state and
+            // stop listening on 'error' event.
+            if (this.options && this.options.autoCreateNamespace) {
+                // for some reason this.client.exists() does not return
+                // NO_NODE when base path does not exist, hence use
+                // getData() instead
+                this.getData('/', err => {
+                    if (err && err.name !== 'NO_NODE') {
+                        this.emit('error', err);
+                        return;
+                    }
+                    // NO_NODE error and autoCreateNamespace is enabled
+                    if (err) {
+                        const nsIndex = this.connectionString.indexOf('/');
```

Comment:
The connection string may not always contain a namespace. Maybe in our products we always have one, but for more genericity it could be better to handle cases where there's no namespace.